File: api/tests/factories/deployments.py
from datetime import datetime, timezone

from deployments.entities import Deployment, DeploymentListItem


def build_deployment_dict(**overrides) -> dict:
    deployment = {
        "deployment_id": "dep-001",
        "version": "1.0.0",
        "status": "active",
        "type": "web_service",
        "environment": "production",
        "attributes": {
            "name": "checkout-api",
            "description": "Handles checkout requests",
            "team": "payments",
            "region": "eu-west-1",
            "language": "python",
            "framework": "fastapi",
            "priority": "high",
            "oncall": "payments-oncall",
        },
        "created_at": datetime(2026, 4, 1, 12, 0, tzinfo=timezone.utc),
        "created_by": "adilson",
        "updated_at": datetime(2026, 4, 10, 9, 30, tzinfo=timezone.utc),
        "deleted_at": None,
    }
    deployment.update(overrides)
    return deployment


def build_deployment(**overrides) -> Deployment:
    return Deployment.model_validate(build_deployment_dict(**overrides))


def build_deployment_list_item(**overrides) -> DeploymentListItem:
    deployment = build_deployment(**overrides)
    return DeploymentListItem(
        deployment_id=deployment.deployment_id,
        attributes=deployment.attributes,
        name=deployment.attributes.name,
        description=deployment.attributes.description,
        team=deployment.attributes.team,
        version=deployment.version,
        status=deployment.status,
        type=deployment.type,
        environment=deployment.environment,
        created_at=deployment.created_at,
        created_by=deployment.created_by,
        updated_at=deployment.updated_at,
    )

File: web/src/test-helpers/deployments.ts
import type { Deployment, DeploymentListItem } from "@/domains/deployments/types";

export function buildDeploymentListItem(
  overrides: Partial<DeploymentListItem> = {},
): DeploymentListItem {
  return {
    attributes: {
      description: "Handles checkout flows",
      framework: "fastapi",
      language: "python",
      name: "checkout-api",
      oncall: "payments-oncall",
      priority: "high",
      region: "eu-west-1",
      team: "payments",
    },
    created_at: "2026-04-01T12:00:00Z",
    created_by: "adilson",
    deployment_id: "dep-001",
    description: "Handles checkout flows",
    environment: "production",
    name: "checkout-api",
    status: "active",
    team: "payments",
    type: "web_service",
    updated_at: "2026-04-10T09:30:00Z",
    version: "1.0.0",
    ...overrides,
  };
}

export function buildDeployment(
  overrides: Partial<Deployment> = {},
): Deployment {
  return {
    attributes: {
      description: "Handles checkout flows",
      framework: "fastapi",
      language: "python",
      name: "checkout-api",
      oncall: "payments-oncall",
      priority: "high",
      region: "eu-west-1",
      team: "payments",
    },
    created_at: "2026-04-01T12:00:00Z",
    created_by: "adilson",
    deleted_at: null,
    deployment_id: "dep-001",
    environment: "production",
    status: "active",
    type: "web_service",
    updated_at: "2026-04-10T09:30:00Z",
    version: "1.0.0",
    ...overrides,
  };
}
